### ISSUE TITLE
yarn 1.0.1 doesn't use custom registry from .npmrc

### ISSUE THREAD

**Issue by tleunen** [yarnpkg/yarn#4421]
**Do you want to request a *feature* or report a *bug*?**
Bug 

**What is the current behavior?**
When having a custom registry in `.npmrc`, yarn doesn't use it and keep using the default npm registry instead. This results in errors during install with:
```
error Couldn't find package "aaa" on the "npm" registry.
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Create a npmrc file with a custom registry:
```
registry = https://another-registry.com
```
Then try to install your dependencies from the custom registry

**What is the expected behavior?**
It worked fine with v0.x. The custom registry was used when fetching the dependencies.

**Please mention your node.js, yarn and operating system version.**
Node 8.4.0
Yarn 1.0.1
Mac OS Sierra

**Comment by BYK** [comment]
The error message is misleading since it just uses the registry id but uses the URL provided underneath. I think you are experiencing an issue that is fixed on master. Would you mind trying one of the nightlies?

**Comment by BYK** [comment]
I'll use this issue to make the error message more accurate if your issue is resolved on master. Thanks for reporting!

**Comment by tleunen** [comment]
Indeed, it worked with the latest nightly.
Do you know when a new version of yarn with this fix will be released?

Thank you.

**Comment by BYK** [comment]
>Do you know when a new version of yarn with this fix will be released?

We are getting ready for it as we speak. It should be out in a few days.

Thanks for verifying!

**Comment by BYK** [comment]
Oh wait, I need to keept this open to fix the error message :)

**Comment by jacob-israel-turner** [comment]
>  I think you are experiencing an issue that is fixed on master. Would you mind trying one of the nightlies?
@BYK Could you link me to this issue?  I think we're experiencing it on 1.0.1.  Yarn doesn't seem to be resolving private modules resolved on the npmjs registry.  I want to see if ya'll have resolved it already.

**Comment by jacob-israel-turner** [comment]
It looks like the nightly build fixes this issue.

**Comment by marcelometal** [comment]
Please, release it ;)

**Comment by BYK** [comment]
>Please, release it ;)

Noo, I'm saving it for myself :P

**Comment by timothystewart6** [comment]
I think this can be closed.  Seems to be fixed in `1.0.2` https://github.com/yarnpkg/yarn/releases/tag/v1.0.2

```

```

**Comment by BYK** [comment]
🎉
